lstchain/tools/lstchain_create_time_calibration_file.py

```diff
@@ -0,0 +1,108 @@
+"""
+Create drs4 time correction coefficients.
+"""
+import glob
+import numpy as np
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe_io_lst import LSTEventSource
+from lstchain.calib.camera.r0 import LSTR0Corrections
+from lstchain.calib.camera.time_correction_calculate import TimeCorrectionCalculate
+
+
+class TimeCalibrationHDF5Writer(Tool):
+
+    name = "TimeCalibrationHDF5Writer"
+    description = "Generate a HDF5 file with time calibration coefficients"
+
+    input_file = traits.Unicode(
+        help="Path to the input file containing events (wildcards allowed)",
+    ).tag(config=True)
+
+    output_file = traits.Unicode(
+        help="Path to the time calibration file",
+    ).tag(config=True)
+
+    pedestal_file = traits.Unicode(
+        help="Path to drs4 pedestal file",
+    ).tag(config=True)
+
+    max_events = traits.Int(
+        help="Maximum numbers of events to read. Default = 20000",
+        default_value=20000
+    ).tag(config=True)
+
+    aliases = {
+        "input_file": "TimeCalibrationHDF5Writer.input_file",
+        "output_file": "TimeCalibrationHDF5Writer.output_file",
+        "pedestal_file": "LSTR0Corrections.pedestal_path",
+        "max_events": "TimeCalibrationHDF5Writer.max_events",
+    }
+
+    classes = [LSTEventSource, LSTR0Corrections, TimeCorrectionCalculate]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a HDF5 file with time calibration coefficients.
+
+        For getting help run:
+        lstchain_create_time_calibration_file --help
+        """
+
+        self.timeCorr = None
+        self.path_list = None
+        self.lst_r0 = None
+
+    def setup(self):
+
+        self.path_list = sorted(glob.glob(self.input_file))
+        reader = LSTEventSource(input_url=self.path_list[0], max_events=self.max_events)
+        self.lst_r0 = LSTR0Corrections(
+            pedestal_path=self.pedestal_file,
+            config=self.config,
+        )
+        self.timeCorr = TimeCorrectionCalculate(
+            calib_file_path=self.output_file,
+            subarray=reader.subarray,
+            config=self.config,
+        )
+
+    def start(self):
+
+        try:
+            for j, path in enumerate(self.path_list):
+                reader = LSTEventSource(input_url=self.path_list[0], max_events=self.max_events)
```

Comment:
Shouldn't the eventsource get `path` as input? Like this you are only looking at the first file.